### PR TITLE
`Development`: Add new MySQL and Postgres migration check

### PR DESCRIPTION
**Pull request by TheZoker**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

We sometimes have issues with migrations not running within postgres. 

### Description

This PR adds a new migration check to bamboo, which will start artemis and MySQL or Postgres and when the health check passes, it will stop the containers. When a issue occurs during migration, bamboo should fail. 

#### Code Review
- [x] Review 1
- [ ] Review 2

### CODE REVIEW

File: .bamboo/migration-check/execute.sh
Comment (MaximilianAnzinger) [inline .bamboo/migration-check/execute.sh:33]: This output seems redundant since you output the exit code anyways.

**Review by MaximilianAnzinger — DISMISSED** [review]
This will make life so much easier 👍 

Looks good to me. I left one optional comment.

File: .bamboo/migration-check/execute.sh
Comment (4ludwig4) [reply .bamboo/migration-check/execute.sh:33]: I guess some other system will check for the `.successful` file? Or is this is about the last line with the `echo`?

File: docker/artemis-migration-check-mysql.yml
Comment (4ludwig4) [inline docker/artemis-migration-check-mysql.yml:11]:
```suggestion
```
Do we need access to the Artemis instance? If not I guess we can delete the ports or limit it to `127.0.0.1`? I guess we just check the exit code anyhow.

File: docker/artemis-migration-check-postgres.yml
Comment (4ludwig4) [inline docker/artemis-migration-check-postgres.yml:11]:
```suggestion
```
Same as above.

**Review by 4ludwig4 — COMMENTED** [review]
Are any cleanup scripts necessary?
Besides that I have some suggestions see the comments.
The bamboo runs seem to be working for this branch.
Is there an integration in the GitHub workflows which checks the status for every branch or just for `develop`?

File: .bamboo/migration-check/execute.sh
Comment (TheZoker) [reply .bamboo/migration-check/execute.sh:33]: I just copied this from the cypress execute script. I'm not sure if bamboo uses this information

File: docker/artemis-migration-check-mysql.yml
Comment (TheZoker) [reply docker/artemis-migration-check-mysql.yml:13]: I'm not sure if it is worth it to create a new `.env` file just for one single line of code 🤔

File: docker/artemis-migration-check-mysql.yml
Comment (TheZoker) [reply docker/artemis-migration-check-mysql.yml:11]: Good point, I will remove it

File: docker/artemis-migration-check-postgres.yml
Comment (maximiliansoelch) [inline docker/artemis-migration-check-postgres.yml:11]: Is this here on purpose?

Comment (TheZoker) [reply docker/artemis-migration-check-postgres.yml:11]: Yes it is needed, so artemis uses the postgres connector instead of the mysql connector. I can rename the file to something more generic (maybe `postgres-dev.env` or something)

Comment (4ludwig4) [reply docker/artemis-migration-check-postgres.yml:11]: @maximiliansoelch, @TheZoker this is already handled [in this PR](https://github.com/ls1intum/Artemis/pull/6759) where I also add the development postgres setups ;)

File: .bamboo/migration-check/execute.sh
Comment (maximiliansoelch) [inline .bamboo/migration-check/execute.sh:34]: As this part already led to some discussions, I suggest you remove the creation of the `.successful` file from here and instead add another task for creating this file to both stages in Bamboo.
This also makes it easier in Bamboo to understand where the .successful file comes from.

```suggestion
```

Comment (TheZoker) [reply .bamboo/migration-check/execute.sh]: Done :)

**Review by maximiliansoelch — APPROVED** [review]
Code looks good now 👍

@ls1intum/artemis-maintainers 
After the merge, polling needs to be enabled in the [Artemis Migration Check](https://bamboo.ase.in.tum.de/browse/ARTEMIS-AMC0) plan on Bamboo.
Further, the branch protection rules should be updated to require successful migration checks for both, MySQL and Postgres.

**Pull request merged**
